rclcpp/include/rclcpp/experimental/subscription_intra_process.hpp

```diff
@@ -115,13 +116,34 @@ class SubscriptionIntraProcess : public SubscriptionIntraProcessBase
   bool
   is_ready(rcl_wait_set_t * wait_set)
   {
-    (void)wait_set;
+    (void) wait_set;
     return buffer_->has_data();
   }
 
-  void execute()
+  void take_data(std::shared_ptr<void> & data)
   {
-    execute_impl<CallbackMessageT>();
+    if (data) {
+      throw std::runtime_error("'data' is not empty");
```

Comment:
Nit: `#include <stdexcept>` for `std::runtime_error`.

Reply:
Done in https://github.com/ros2/rclcpp/pull/1241/commits/5b39f357bbe0096af397b0b782542452153ac517.